### PR TITLE
Bookmarked for user

### PR DESCRIPTION
**Pull request by joshmorel**
This now works for both `/users/me/bookmarked` and `/user/{userId}/bookmarked` and is up-to-date with develop

### CODE REVIEW

**Comment by jasonify** [comment]
Thanks for finishing this up.

I am going to be a little busy with the username bug / problem so I'll make sure to take a close look after that's completed.

File: server/tests/favorite.test.js [hunk 1]
```diff
@@ -258,4 +260,50 @@ describe('## Favorite APIs', () => {
...
+          .set('Authorization', `Bearer ${userToken}`)
+          .expect(httpStatus.OK))
+        .then((res) => {
+          expect(res.body).to.be.an('array');
+          expect(res.body[0]._id).to.equal(`${postId}`);
+          done();
+        })
+        .catch(done);
+    });
+    it('should not get non-bookmarked for authenticated user with "me" shortcut', (done) => {
```
Comment (jasonify) [inline server/tests/favorite.test.js]: :) can't think of a better way to word this but it confused me for a second.

**Comment by jasonify** [comment]
This looks good, though it looks like there were some conflicts, I think it just has to do with formatting things?

**Comment by joshmorel** [comment]
Looks like it. If you keep the either branch's version I really don't think there would be any issues.

**Pull request merged**